python/pyarrow/_flight.pyx

```diff
@@ -2550,11 +2577,13 @@ cdef class FlightServerBase(_Weakrefable):
     """
 
     cdef:
-        unique_ptr[PyFlightServer] server
+        shared_ptr[PyFlightServer] server
+        object _finalizer
```

Comment:
Nit, but I don't think `cdef` members are exposed in Python, so could name it `finalizer`?